src/img_boss/make_vdi.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+#~ Copyright (C) 2013 David Greaves <david@dgreaves.com>
+#~ This program is free software: you can redistribute it and/or modify
+#~ it under the terms of the GNU General Public License as published by
+#~ the Free Software Foundation, either version 3 of the License, or
+#~ (at your option) any later version.
+
+#~ This program is distributed in the hope that it will be useful,
+#~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+#~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#~ GNU General Public License for more details.
+
+#~ You should have received a copy of the GNU General Public License
+#~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Converts mic raw image to VDI format.
+
+.. warning ::
+
+   * The make_vdi participant must run on the master nfs share
+   * Must have run after build_image and relies on a valid
+     build_image workitem
+
+:term:`Workitem` fields IN:
+
+:Parameters:
+
+   :vboxmanage.process (Boolean):
+      If false then the participant is skipped
+   :vboxmanage.format (string):
+      As per VBoxManage --format; one of VDI|VMDK|VHD
+   :vboxmanage.variant (string):
+      OPTIONAL As per VBoxManage --variant; one or more of:
+        Standard,Fixed,Split2G,Stream,ESX
+
+:term:`Workitem` fields OUT:
+
+:Returns:
+  :result (Boolean):
+     True if everything was OK, False otherwise
+"""
+
+from RuoteAMQP.workitem import DictAttrProxy as dap
+from RuoteAMQP import Launcher 
+
+import os.path
+import subprocess
+import shutil
+
+class ParticipantHandler(object):
+    """Participant class as defined by the SkyNET API"""
+
+    def __init__(self):
+        self.worker_config = None
+
+    def handle_wi_control(self, ctrl):
+        """ job control thread """
+        pass
+
+    def handle_lifecycle_control(self, ctrl):
+        """ participant control thread """
+        if ctrl.message == "start":
+            if ctrl.config.has_section("make_vdi"):
+                self.config = dap({})
+                self.config.base_url = ctrl.config.get("make_vdi", "base_url")
+                self.config.base_dir = ctrl.config.get("make_vdi", "base_dir")
+                print "config base_url %s" % self.config.base_url
+
+    def handle_wi(self, wid):
+        """Handle the workitem to convert an image
+        """
+        f = wid.fields
+
+        if not f.image or not f.image.image_type or not f.image.image_url:
+            raise RuntimeError("Missing mandatory field 'image'")
+
+        if not f.image.image_type == "raw":
+            return False, "Image type is not raw"
+
+        raw = os.path.abspath(f.image.image_url.replace(self.config.base_url,
```

Comment:
Could this instead use

    os.path.join(f.image.outdir, os.path.basename(f.image.image_url))

so there wouldn't be need for configuration?

Though I'm not sure if outdir is intentionally included in the image data that ends up in the WI

Reply:
I think there's some path data too - like web/lbt/ ??

Reply:
it seems that f.image.outdir is not set and the outdir isn't in fields so :(